toolchain/parse/typed_nodes_test.cpp

```diff
@@ -137,6 +137,149 @@ TEST_F(TypedNodeTest, For) {
   ASSERT_TRUE(for_var_name.has_value());
 }
 
+TEST_F(TypedNodeTest, VerifyExtractTraceLibrary) {
+  auto* tree = &GetTree(R"carbon(
+    library default impl;
+  )carbon");
+  auto file = tree->ExtractFile();
+
+  ASSERT_EQ(file.decls.size(), 1);
+  ErrorBuilder trace;
+  auto library = tree->VerifyExtractAs<LibraryDirective>(file.decls[0], &trace);
+  EXPECT_TRUE(library.has_value());
+  Error err = trace;
+  // Use Regex matching to avoid hard-coding the result of `typeinfo(T).name()`.
+  EXPECT_THAT(err.message(), testing::MatchesRegex(
+                                 R"Trace(Aggregate [^:]*: begin
```

Comment:
I think this would also work and might be a bit more readable?
```suggestion
                                 R"Trace(\
Aggregate [^:]*: begin
```

(same pattern below if it does work)

Reply:
Doesn't work.